app/routes/quickstart/credentials.tsx

```diff
@@ -6,248 +6,22 @@
  * SPDX-License-Identifier: NASA-1.3
  */
 
-import type { ModalRef } from '@trussworks/react-uswds'
-import {
-  Button,
-  Dropdown,
-  Grid,
-  Label,
-  Modal,
-  ModalFooter,
-  ModalHeading,
-  ModalToggleButton,
-  TextInput,
-} from '@trussworks/react-uswds'
-import { Form, Link, useFetcher, useLoaderData } from '@remix-run/react'
-import type { RedactedClientCredential } from '../user/client_credentials.server'
-import { ClientCredentialVendingMachine } from '../user/client_credentials.server'
+import { useLoaderData } from '@remix-run/react'
 import type { DataFunctionArgs } from '@remix-run/node'
-import { redirect } from '@remix-run/node'
-import moment from 'moment'
-import type { ReactNode } from 'react'
-import { useRef, useState } from 'react'
-import ReCAPTCHA from 'react-google-recaptcha'
-import { getEnvOrDieInProduction } from '~/lib/env'
+import SegmentedCards from '~/components/SegmentedCards'
+import {
+  handleCredentialActions,
+  handleCredentialLoader,
+  NewCredentialForm,
+} from '~/components/NewCredentialForm'
+import CredentialCard from '~/components/CredentialCard'
 
 export async function loader({ request }: DataFunctionArgs) {
-  const machine = await ClientCredentialVendingMachine.create(request)
-  const client_credentials = await machine.getClientCredentials()
-  const groups = machine.groups
-  const recaptchaSiteKey = getEnvOrDieInProduction('RECAPTCHA_SITE_KEY')
-  return { client_credentials, recaptchaSiteKey, groups }
-}
-
-async function verifyRecaptcha(response?: string) {
-  const secret = getEnvOrDieInProduction('RECAPTCHA_SITE_SECRET')
-  if (!secret) return
-
-  const params = new URLSearchParams()
-  if (response) {
-    params.set('response', response)
-  }
-  params.set('secret', secret)
-  const verifyResponse = await fetch(
-    'https://www.google.com/recaptcha/api/siteverify',
-    { method: 'POST', body: params }
-  )
-  const { success } = await verifyResponse.json()
-  if (!success) throw new Response('ReCAPTCHA was invalid', { status: 400 })
-}
-
-function getFormDataString(formData: FormData, key: string) {
-  const value = formData.get(key)
-  if (typeof value === 'string') {
-    return value
-  } else if (value === null) {
-    return undefined
-  } else {
-    throw new Response(`expected ${key} to be a string`, { status: 400 })
-  }
+  return await handleCredentialLoader({ request } as DataFunctionArgs)
```

Comment:
The cast should not be necessary:

```suggestion
  return await handleCredentialLoader({ request })
```